File: produto-otear-os/sistemas-prontos/otear-prospeccao/apps/o-tear-prospeccao/backend/app/schemas/lead.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum
import uuid


class LeadClassification(str, Enum):
    HOT = "hot"
    WARM = "warm"
    COLD = "cold"


class OutreachStatus(str, Enum):
    IMPORTED = "imported"
    QUEUED = "queued"
    CONTACTED = "contacted"
    REPLIED = "replied"
    CONVERTED = "converted"
    OPTED_OUT = "opted_out"
    ARCHIVED = "archived"


class LeadResponse(BaseModel):
    id: str
    prospect_pro_id: Optional[str] = None
    name: str
    category: Optional[str] = None
    phone: Optional[str] = None
    website: Optional[str] = None
    address: Optional[str] = None
    rating: Optional[float] = None
    reviews_count: Optional[int] = None
    maps_url: Optional[str] = None
    instagram_handle: Optional[str] = None
    instagram_followers: Optional[int] = None
    engagement_rate: Optional[float] = None
    cnpj: Optional[str] = None
    razao_social: Optional[str] = None
    nome_fantasia: Optional[str] = None
    decision_maker_name: Optional[str] = None
    decision_maker_role: Optional[str] = None
    cnpj_owners: list[dict] = []
    score_total: int = 0
    score_classification: Optional[str] = None
    score_bad_website: int = 0
    site_status: Optional[str] = None
    site_score: Optional[int] = None
    site_final_url: Optional[str] = None
    site_problems: list[str] = []
    site_missing_items: list[str] = []
    site_present_items: list[str] = []
    site_response_time_seconds: Optional[float] = None
    site_http_status: Optional[int] = None
    site_audit: dict = {}
    approach_script: Optional[str] = None
    outreach_status: str = "imported"
    whatsapp_number: Optional[str] = None
    tags: list[str] = []
    notes: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class LeadListResponse(BaseModel):
    total: int
    hot: int
    warm: int
    cold: int
    leads: list[LeadResponse]


class LeadFilters(BaseModel):
    classification: Optional[str] = None
    outreach_status: Optional[str] = None
    min_score: Optional[int] = None
    tags: Optional[list[str]] = None
    search: Optional[str] = None
    page: int = 1
    per_page: int = 25


class BulkActionRequest(BaseModel):
    lead_ids: list[str]
    action: str  # "tag", "archive", "assign_campaign"
    value: Optional[str] = None  # tag name or campaign_id


class ImportResponse(BaseModel):
    batch_id: str
    total_imported: int
    duplicates_skipped: int
    updated_existing: int = 0
    errors: int
    error_details: list[str] = []

File: produto-otear-os/sistemas-prontos/otear-prospeccao/apps/o-tear-prospeccao/frontend/src/types/lead.ts
export interface Lead {
  id: string
  prospect_pro_id?: string
  name: string
  category?: string
  phone?: string
  website?: string
  address?: string
  rating?: number
  reviews_count?: number
  maps_url?: string
  latitude?: number
  longitude?: number
  instagram_handle?: string
  instagram_url?: string
  instagram_bio?: string
  instagram_bio_link?: string
  instagram_followers?: number
  instagram_following?: number
  instagram_posts?: number
  instagram_is_business?: boolean
  instagram_is_verified?: boolean
  engagement_rate?: number
  avg_likes?: number
  avg_comments?: number
  posting_frequency_days?: number
  last_post_date?: string
  cnpj?: string
  razao_social?: string
  nome_fantasia?: string
  decision_maker_name?: string
  decision_maker_role?: string
  cnpj_owners?: Array<{
    name?: string
    role?: string
  }>
  score_total: number
  score_classification?: 'hot' | 'warm' | 'cold'
  score_no_website?: number
  score_bad_website?: number
  score_no_instagram?: number
  score_low_engagement?: number
  score_irregular_posting?: number
  score_few_reviews?: number
  score_low_rating?: number
  score_no_professional_bio?: number
  score_no_bio_link?: number
  site_status?: 'missing' | 'bad' | 'ok' | string
  site_score?: number
  site_final_url?: string
  site_problems?: string[]
  site_missing_items?: string[]
  site_present_items?: string[]
  site_response_time_seconds?: number
  site_http_status?: number
  site_audit?: {
    ads?: {
      status?: 'tracking_detected' | 'analytics_only' | 'unknown'
      has_tracking_pixels?: boolean
      has_paid_ads_signals?: boolean
      platforms?: string[]
      signals?: string[]
      meta_ads_library_url?: string
      google_ads_transparency_url?: string
      domain?: string
      note?: string
    }
    [key: string]: unknown
  }
  approach_script?: string
  outreach_status: string
  whatsapp_number?: string
  tags: string[]
  notes?: string
  created_at?: string
  updated_at?: string
}

export interface LeadListResponse {
  total: number
  hot: number
  warm: number
  cold: number
  leads: Lead[]
}

export interface ImportResponse {
  batch_id: string
  total_imported: number
  duplicates_skipped: number
  updated_existing?: number
  errors: number
  error_details: string[]
}
